### DISCUSSION TITLE
Idea: add Option to set filetype based on content-type

### DISCUSSION THREAD

**Discussion by kevintraver** [rest-nvim/rest.nvim#231] [Ideas]
Idea: Add an option to update the _buffer_ **filetype** to match the **content-type** (instead of `httpResult`).

This could be an opt-in feature (not default).

For example, if the response was `json`, the buffer filetype would be set to `json`. Could maybe also default to `html`.

This would be awesome for things like treesitter.

**Comment by kevintraver** [comment]
Another idea: have an option to split the results panel window, and have one buffer be the headers with type `httpResult`, and the body would be a second buffer with whatever the returned _content-type_ is (`html`, `json`, etc...).

**Reply by boltlessengineer** [reply]
This is now possible with v3 release.
Each panes in response window has dedicated buffers, so you can open multiple response views at once.
For example, you can open *Response* pane and *Headers* pane in same time like screenshot from README.

**Comment by boltlessengineer** [comment]
rest.nvim now use `tree-sitter-http` parser for response UI highlights and `http` parser supports general injections for json, xml and graphql.

So you can use all tree-sitter features from response pane. (e.g. highlights, injections, folds)
